searx/engines/mastodon.py

```diff
@@ -0,0 +1,75 @@
+# SPDX-License-Identifier: AGPL-3.0-or-later
+# lint: pylint
+"""Mastodon is an open source alternative to large social media platforms
+like Twitter/X, Facebook, ...
+
+Since it's federated and self-hostable, there's a large amount of available
+instances, which can be chosen instead by modifying ``base_url``.
+
+We use their official API for searching, but unfortunately, their Search API
+forbids pagination without OAuth.
+
+That's why we use tootfinder.ch for finding posts, which doesn't support searching
+for users, accounts or other types of content on Mastodon however.
+"""
+
+from urllib.parse import urlencode
+from datetime import datetime
+
+about = {
+    "website": 'https://joinmastodon.org/',
+    "wikidata_id": 'Q27986619',
+    "official_api_documentation": 'https://docs.joinmastodon.org/api/',
+    "use_official_api": True,
+    "require_api_key": False,
+    "results": 'JSON',
+}
+categories = ['social media']
+
+base_url = "https://mastodon.social"
+mastodon_type = "accounts"
+
+page_size = 10
+
+
+def request(query, params):
+    args = {
+        'q': query,
+        'resolve': 'false',
+        'type': mastodon_type,
+        'limit': page_size,
+    }
+    params['url'] = f"{base_url}/api/v2/search?{urlencode(args)}"
```

Comment:
Is there no page argument / does the API support paging?

Reply:
Only when using OAuth, see
https://github.com/searxng/searxng/pull/2857/files#diff-448f76c6b4dc52de868f8b33e441e68d9e34d13caa24c6640c7edeb0eff5c1c7R10

You can see the error messages popping up when trying to scroll down in the search results on mastodon.social without an account.

Reply:
Ooops,sorry, my eyes were probably already too tired .. :+1: